Cookbook/Technical-Documents/CRP-Bot.md

```diff
@@ -0,0 +1,99 @@
+
+Again, this might seem surprising at first, but this is expected and on purpose.
+
+### Impact on "Fix versions" vs feature flags
+
+Another impact of that process is that **even features that are gated by a local feature switch** (hardcoded as being turned off in code) and end up still being disabled when we release the app **will still be flagged with the released version in their "Fix Version"**.
+
+This is because – similar to what was described above with the example of unwanted side effects of a US feature on the UK app – the implementation of a feature that is theoretically supposed to be disabled by a feature flag can accidentally introduce bugs or have some of the code not protected behind the feature flag like it should have been. So since the code is present in the final app anyway – even if it's supposed to be disabled – the corresponding JIRA ticket will still have the version set in its "Fix Version".
+
+This isn't really an issue in the end since usually we'll end up having a separate ticket to enable the feature switch and the feature in the future. So that new JIRA ticket about turning the feature on will have its own "Fix Version" telling us when it was enabled.
+
+Still, this is a behavior to be aware of to understand the real meaning we put behind that "Fix Version" field (i.e. "the code implementing this feature ended up in the codebase shipped on the store for this version, even if supposedly disabled").
+
+### "Fix versions" and planning versions ahead
+
+Some squads might want to use JIRA releases to plan ahead, at the beginning of a sprint, which tickets they hope to be part of a future release.
+
+Since there's no guarantee that those tickets will actually make it in time in the planned release when the corresponding release branch is cut, we need to differentiate a ticket that is just _planned_ from a ticket that _actually made that release_.
+
+As JIRA releases handled by the CRP bot are dedicated to mark tickets that _actually shipped_ in a release (i.e. that are in the corresponding release branch), **you should never use those `"<platform> <app> <version>"` JIRA releases to mark planned tickets** via the `Fix Version` field. Otherwise that will mess up with our SSDLC requirement about tracking which tickets actually shipped with which version.
+
+Instead, you should use JIRA Versions with a name that doesn't conflict with the ones used by the bot.  
+We recommend naming those JIRA releases for _planned_ roadmaps something like **"<Platform> <Flavor> RC <Version>"** – e.g. "iOS Babylon RC 4.10.0". "RC" here meaning "Release Candidate", it will help differentiate that kind of planned JIRA versions from the version used by the bot when the ticket is actually released.
```

Comment:
```suggestion
We recommend naming those JIRA releases for _planned_ roadmaps something like **"<Platform> <Flavor> RC <Version>"** – e.g. "iOS Babylon RC 4.10.0". "RC" here meaning "Release Candidate", it will help differentiate that kind of planned JIRA releases from the JIRA release used by the bot when the ticket is actually released.
```